File: array/js/third-largest-element.js
const getThirdLargestElement = (arr) => {
  let firstLargest = -1, secondLargest = -1, thirdLargest = -1;

  for (const e in arr) {
    if (arr[e] > firstLargest) {
      thirdLargest = secondLargest;
      secondLargest = firstLargest;
      firstLargest = arr[e];
    } else if (arr[e] > secondLargest && arr[e] < firstLargest) {
      thirdLargest = secondLargest;
      secondLargest = arr[e];
    } else if (arr[e] > thirdLargest && arr[e] < secondLargest) {
      thirdLargest = arr[e];
    }
  }

  console.log('Third largest:', thirdLargest);
  return thirdLargest;
};

const arr = [12, 35, 1, 10, 34, 1];
console.log(getThirdLargestElement(arr));

File: array/rust/src/third_largest_element.rs
pub fn get_third_largest(arr: &[i32]) -> i32 {
    let mut largest = -1;
    let mut second_largest = -1;
    let mut third_largest = -1;

    for e in arr {
        if e > &largest {
            third_largest = second_largest;
            second_largest = largest;
            largest = *e;
        } else if e > &second_largest {
            third_largest = second_largest;
            second_largest = *e;
        } else if e > &third_largest {
            third_largest = *e;
        }
    }

    println!("Third largest element: {}", third_largest);

    third_largest
}
